src/decompression.jl

```diff
@@ -0,0 +1,59 @@
+module DecompressorStreams
+    using CodecZlib, Requires
+    export ArchiveFormat, decompressor_stream
+
+    abstract type ArchiveFormat{S} end
+
+    err_message(name) = string(
+        "Codec$name package is required to read $name-compressed RData files. ",
+        "Run Pkg.add(\"Codec$name\") to install it. Then in your code call ",
+        "using Codec$name before calling using RData."
+    )
+
+    function decompressor_stream(::Type{ArchiveFormat{S}}, io::IOStream) where S
+        if S == :bzip2
+            error(err_message("Bzip2"))
```

Comment:
```suggestion
            error(codec_missing_msg(string(S))
```

Maybe then we even don't need message-generating function. The format symbols has to be updated later in the code (`:gzip` > `:Gzip` etc)